### PR TITLE
mergeAccounts removes duplicate emails and displayNames

### PR DESCRIPTION
**Pull request by Raemon**
People often end up with duplicate accounts sharing an email, and it'd be helpful if we could merge those in a way that actually removes the duplication so they can then actually edit their accounts without throwing validation errors.

This PR also makes it so that the mergeAccounts function specifies sourceUserId and targetUserId, because I frequently had to go looking up which was which, which I found annoying. (I personally access the command by searching my command line history, and grab the last instance of it). So, now it looks like:

```
scripts/serverShellCommand.sh "Globals.mergeAccounts({sourceUserId:'vperH823EjMoAnzAz', targetUserId:'bneALscHnCXsr8QRu', dryRun:false})"
```

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203050878789260) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by Raemon** [comment]
(I have tested this, except for the duplicate displayNames functionality – I couldn't actually cause a duplicate display name, maybe we don't need to worry about it)

**Comment by jpaddison3** [comment]
Sorry for the delay — I've been pretty buried, for which I apologize.

File: packages/lesswrong/server/scripts/mergeAccounts.ts [hunk 1]
```diff
@@ -338,14 +342,60 @@ Vulcan.mergeAccounts = async (sourceUserId: string, targetUserId: string, dryRun
...
+          }}
+        );
+      }
+      const sourceEmailsEmail = (sourceUser.emails.length > 0) && sourceUser.emails[0]
+      const targetEmailsEmail = (targetUser.emails.length > 0) && targetUser.emails[0]
+      if (sourceEmailsEmail === targetEmailsEmail) {
+        await Users.rawUpdateOne(
+          {_id: sourceUserId},
+          {$set: {
+            'emails.0': newEmail
```
Comment (jpaddison3) [inline packages/lesswrong/server/scripts/mergeAccounts.ts:371]: {
```suggestion
            'emails.0': {address: newEmail, verified: source.emails[0].verified}
```

File: packages/lesswrong/server/scripts/mergeAccounts.ts [hunk 2]
```diff
@@ -338,14 +342,60 @@ Vulcan.mergeAccounts = async (sourceUserId: string, targetUserId: string, dryRun
...
+        await Users.rawUpdateOne({_id: sourceUserId}, {$set: { email: newDisplayName}}
+        );
+      }
+    }
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.log("%c Error changing displayName", "color: red")
+    // eslint-disable-next-line no-console
+    console.log(err)
+  }
```
Comment (jpaddison3) [inline packages/lesswrong/server/scripts/mergeAccounts.ts:397]: [nit] I'd move this to be right after the slug change

File: scripts/mergeUsers.sh [hunk 3]
```diff
@@ -33,4 +33,4 @@ TARGET_ID=${@:$OPTIND+1:1}
 
 echo "Merging user $SOURCE_ID into $TARGET_ID"
 
-scripts/serverShellCommand.sh "Vulcan.mergeAccounts('$SOURCE_ID', '$TARGET_ID', false)"
+scripts/serverShellCommand.sh "Vulcan.mergeAccounts({sourceUserId:'$SOURCE_ID', targetUserId:'$TARGET_ID', dryRun:false})"
```
Comment (jpaddison3) [inline scripts/mergeUsers.sh:36]: Nice, I'm glad you found this.

**Review by jpaddison3 — APPROVED** [review]
Found a bug, but it's easy to address. Otherwise, good to go.

**Pull request merged**